src/components/modebar/buttons.js

```diff
@@ -183,7 +183,8 @@ function handleCartesian(gd, ev) {
         var mag = (val === 'in') ? 0.5 : 2,
             r0 = (1 + mag) / 2,
             r1 = (1 - mag) / 2,
-            axList = Axes.list(gd, null, true);
+            axList = Axes.list(gd, null, true),
+            allEnabled = 'on';
```

Comment:
this (and `fullLayout._cartesianSpikesEnabled`) reads to me like a boolean, and it seems like that would simplify some of the logic too.